gribscan/gribscan.py

```diff
@@ -446,8 +460,13 @@ def build_refs(messages, global_attrs, coords, varinfo, magician):
                 "_ARRAY_DIMENSIONS": list(info["dims"]) + list(info["data_dims"]),
             }
         )
-        shape = [len(coords[dim]) for dim in info["dims"]] + list(info["data_shape"])
-        chunks = [1 for _ in info["shape"]] + list(info["data_shape"])
+        data_shape = info["data_shape"]
+        if data_shape == "__from_data_dims__":
+            if not "data_dims" in info:
+                raise ValueError("To use __from_data_dims__, data_dims must be set")
+            data_shape = list(len(coords[dim]) for dim in info["data_dims"])
+        shape = [len(coords[dim]) for dim in info["dims"]] + data_shape
+        chunks = [1 for _ in info["shape"]] + data_shape
```

Comment:
Would it be possible to shift this to the end of `inspect_grib_indices()`? This would require to modify `varinfo` in place which, admittedly, could be slightly more work. However, our reasoning behind this is that `inspect_grib_indices` would then return a "complete" picture of the dataset with all data shapes being present according to the chosen magician. In the current implementation, some part of the "inspection" actually leaks into the `build_refs` scope which works but blurs the lines between both functions.

Please let me know if the explanation is not clear or you disagree.